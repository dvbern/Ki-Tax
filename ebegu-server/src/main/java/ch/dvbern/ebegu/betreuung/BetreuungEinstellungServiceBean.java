/*
 * Copyright (C) 2024 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.betreuung;

import javax.ejb.Stateless;
import javax.inject.Inject;

import ch.dvbern.ebegu.entities.Betreuung;
import ch.dvbern.ebegu.services.EinstellungService;

import static ch.dvbern.ebegu.enums.EinstellungKey.ANWESENHEITSTAGE_PRO_MONAT_AKTIVIERT;
import static ch.dvbern.ebegu.enums.EinstellungKey.GEMEINDE_MAHLZEITENVERGUENSTIGUNG_ENABLED;
import static ch.dvbern.ebegu.enums.EinstellungKey.SCHULERGAENZENDE_BETREUUNGEN;
import static java.util.Objects.requireNonNull;

@Stateless
public class BetreuungEinstellungServiceBean implements BetreuungEinstellungenService {

	@Inject
	private EinstellungService einstellungService;

	@Override
	public BetreuungEinstellungen getEinstellungen(Betreuung betreuung) {
		return BetreuungEinstellungen.builder()
			.betreuteTageEnabled(isBetreuteTageEnabled(betreuung))
			.schulergaenzendeBetreuungEnabled(showSchulergaenzenschulergaenzendeBetreuungEnabled(betreuung))
			.mahlzeitenVerguenstigungEnabled(einstellungService.isEnabled(GEMEINDE_MAHLZEITENVERGUENSTIGUNG_ENABLED, betreuung))
			.build();
	}

	private boolean isBetreuteTageEnabled(Betreuung betreuung) {
		return betreuung.isAngebotTagesfamilien()
			&& einstellungService.isEnabled(ANWESENHEITSTAGE_PRO_MONAT_AKTIVIERT, betreuung);
	}

	private boolean showSchulergaenzenschulergaenzendeBetreuungEnabled(Betreuung betreuung) {
		return requireNonNull(betreuung.getKind().getKindJA().getEinschulungTyp()).isEingeschult()
			&& einstellungService.isEnabled(SCHULERGAENZENDE_BETREUUNGEN, betreuung);
	}
}
