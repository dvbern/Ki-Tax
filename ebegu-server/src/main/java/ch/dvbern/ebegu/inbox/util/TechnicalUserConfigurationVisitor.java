/*
 * Copyright (C) 2024 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.inbox.util;

import ch.dvbern.ebegu.util.mandant.MandantIdentifier;
import ch.dvbern.ebegu.util.mandant.MandantVisitor;

public class TechnicalUserConfigurationVisitor implements MandantVisitor<TechnicalUserConfiguration> {

	public TechnicalUserConfiguration process(MandantIdentifier mandant) {
		return mandant.accept(this);
	}

	@Override
	public TechnicalUserConfiguration visitBern() {
		return new TechnicalUserConfiguration(
			"88888888-2222-2222-2222-222222222222",
			"99999999-2222-2222-2222-222222222222"
		);
	}

	@Override
	public TechnicalUserConfiguration visitLuzern() {
		return new TechnicalUserConfiguration(
			"88888888-2224-2222-2222-222222222222",
			"99999999-2224-2222-2222-222222222222"
		);	}

	@Override
	public TechnicalUserConfiguration visitSolothurn() {
		return new TechnicalUserConfiguration(
			"88888888-2225-2222-2222-222222222222",
			"99999999-2225-2222-2222-222222222222"
		);
	}

	@Override
	public TechnicalUserConfiguration visitAppenzellAusserrhoden() {
		return new TechnicalUserConfiguration(
			"88888888-2223-2222-2222-222222222222",
			"99999999-2223-2222-2222-222222222222"
		);
	}

	@Override
	public TechnicalUserConfiguration visitSchwyz() {
		return new TechnicalUserConfiguration(
			"88888888-2226-2222-2222-222222222222",
			"99999999-2226-2222-2222-222222222222"
		);
	}
}
