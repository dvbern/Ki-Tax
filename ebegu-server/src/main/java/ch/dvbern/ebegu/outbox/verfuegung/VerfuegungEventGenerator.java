/*
 * Copyright (C) 2019 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.outbox.verfuegung;

import java.util.concurrent.TimeUnit;

import javax.annotation.security.RunAs;
import javax.ejb.Schedule;
import javax.ejb.Stateless;
import javax.inject.Inject;

import ch.dvbern.ebegu.enums.UserRoleName;
import ch.dvbern.ebegu.outbox.EventGeneratorServiceBean;
import ch.dvbern.ebegu.services.AbstractBaseService;
import org.jboss.ejb3.annotation.TransactionTimeout;

@Stateless
@RunAs(UserRoleName.SUPER_ADMIN)
public class VerfuegungEventGenerator extends AbstractBaseService {
	@Inject
	private EventGeneratorServiceBean eventGeneratorServiceBean;
	/**
	 * Each new Verfuegung is published to Kafka via the outbox event system. However, there are already Verfuegungn
	 * in the database which have not been published, because the outbox event system has been added later. Thus,
	 * fetch all these Verfuegungen and publish them once.
	 */
	@TransactionTimeout(value = 3, unit = TimeUnit.HOURS)
	@Schedule(info = "Migration-aid, pushes already existing Verfuegungen to outbox", hour = "5", persistent = true)
	public void migrate() {
		eventGeneratorServiceBean.exportVerfuegungEvent();
	}
}
