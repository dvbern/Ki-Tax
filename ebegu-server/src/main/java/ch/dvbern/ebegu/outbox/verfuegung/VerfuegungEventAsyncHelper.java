/*
 * Copyright (C) 2019 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.outbox.verfuegung;

import java.util.Optional;

import javax.annotation.Resource;
import javax.ejb.Asynchronous;
import javax.ejb.Stateless;
import javax.ejb.TransactionAttribute;
import javax.ejb.TransactionAttributeType;
import javax.enterprise.event.Event;
import javax.inject.Inject;
import javax.transaction.TransactionSynchronizationRegistry;

import ch.dvbern.ebegu.entities.Verfuegung;
import ch.dvbern.ebegu.outbox.ExportedEvent;
import ch.dvbern.lib.cdipersistence.Persistence;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Stateless
public class VerfuegungEventAsyncHelper {

	private static final Logger LOG = LoggerFactory.getLogger(VerfuegungEventAsyncHelper.class);

	@Resource
	private TransactionSynchronizationRegistry txReg;

	@Inject
	private Persistence persistence;

	@Inject
	private Event<ExportedEvent> event;

	@Inject
	private VerfuegungEventConverter verfuegungEventConverter;

	@Asynchronous
	@TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
	public void convert(String id) {
		Verfuegung verfuegung = persistence.find(Verfuegung.class, id);

		LOG.info(
			"Converting {} in Thread {} and Transaction {}",
			verfuegung.getBetreuung().getBGNummer(),
			Thread.currentThread(),
			txReg.getTransactionKey());

		Optional<VerfuegungVerfuegtEvent> eventOpt = verfuegungEventConverter.of(verfuegung);

		eventOpt.ifPresent(verfuegungVerfuegtEvent -> {
			this.event.fire(verfuegungVerfuegtEvent);
			verfuegung.setSkipPreUpdate(true);
			verfuegung.setEventPublished(true);
			persistence.merge(verfuegung);
		});
	}
}
