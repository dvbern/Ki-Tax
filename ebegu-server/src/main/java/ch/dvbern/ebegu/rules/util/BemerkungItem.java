/*
 * Copyright (C) 2019 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.rules.util;

import java.util.Objects;

import javax.annotation.Nonnull;

import ch.dvbern.ebegu.types.DateRange;

public class BemerkungItem implements Comparable<BemerkungItem> {

	private final DateRange range;
	private final String message;

	public BemerkungItem(DateRange range, String message) {
		this.range = range;
		this.message = message;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		if (!(o instanceof BemerkungItem)) {
			return false;
		}
		BemerkungItem that = (BemerkungItem) o;
		return Objects.equals(range, that.range) &&
			Objects.equals(message, that.message);
	}

	@Override
	public int hashCode() {
		return Objects.hash(range, message);
	}

	@Override
	public int compareTo(@Nonnull BemerkungItem otherItem) {
		return this.getRange().compareTo(otherItem.getRange());
	}

	public DateRange getRange() {
		return range;
	}

	public String getMessage() {
		return message;
	}
}
