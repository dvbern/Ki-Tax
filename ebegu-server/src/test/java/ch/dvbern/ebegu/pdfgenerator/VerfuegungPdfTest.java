/*
 * Copyright (C) 2024 DV Bern AG, Switzerland
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ch.dvbern.ebegu.pdfgenerator;

import java.io.File;
import java.io.IOException;
import java.math.BigDecimal;
import java.nio.file.Files;
import java.nio.file.Path;
import java.time.LocalDate;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.regex.Pattern;

import javax.annotation.Nonnull;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import ch.dvbern.ebegu.entities.Betreuung;
import ch.dvbern.ebegu.entities.Gemeinde;
import ch.dvbern.ebegu.entities.Gesuch;
import ch.dvbern.ebegu.entities.GemeindeStammdaten;
import ch.dvbern.ebegu.entities.Gesuchsperiode;
import ch.dvbern.ebegu.entities.Mandant;
import ch.dvbern.ebegu.entities.Verfuegung;
import ch.dvbern.ebegu.entities.VerfuegungZeitabschnitt;
import ch.dvbern.ebegu.enums.betreuung.BetreuungspensumAnzeigeTyp;
import ch.dvbern.ebegu.enums.betreuung.Betreuungsstatus;
import ch.dvbern.ebegu.pdfgenerator.verfuegung.AbstractVerfuegungPdfGenerator.Art;
import ch.dvbern.ebegu.pdfgenerator.verfuegung.VerfuegungPdfGeneratorKonfiguration;
import ch.dvbern.ebegu.pdfgenerator.verfuegung.VerfuegungPdfGeneratorVisitor;
import ch.dvbern.ebegu.test.TestDataUtil;
import ch.dvbern.ebegu.test.mandant.MandantFactory;
import ch.dvbern.ebegu.test.util.TestDataInstitutionStammdatenBuilder;
import ch.dvbern.ebegu.testfaelle.Testfall01_WaeltiDagmar;
import ch.dvbern.ebegu.tests.util.PdfUnitTestUtil;
import ch.dvbern.ebegu.types.DateRange;
import ch.dvbern.ebegu.util.Constants;
import ch.dvbern.ebegu.util.mandant.MandantIdentifier;
import ch.dvbern.lib.invoicegenerator.errors.InvoiceGeneratorException;
import org.apache.commons.io.FileUtils;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;
import org.junit.jupiter.params.provider.EnumSource.Mode;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.stringContainsInOrder;

@ExtendWith(SnapshotExtension.class)
class VerfuegungPdfTest {

	private static final String PATH_PREFIX = FileUtils.getTempDirectoryPath() + "/kiBon/Verfuegung/";
	private static final Pattern PATTERN = Pattern.compile("\\r");

	@SuppressWarnings({ "unused", "InstanceVariableMayNotBeInitialized" })
	private Expect expect;

	@BeforeAll
	static void beforeAll() throws IOException {
		FileUtils.deleteDirectory(new File(PATH_PREFIX));

		Arrays.stream(MandantIdentifier.values())
			.map(VerfuegungPdfTest::getOutputPath)
			.forEach(path -> {
				//noinspection ResultOfMethodCallIgnored
				path.toFile().mkdirs();
			});
	}

	@SnapshotName("verfuegung-bruennen")
	@ParameterizedTest
	@EnumSource(value = MandantIdentifier.class, mode = Mode.MATCH_ALL)
	void testBruennen(MandantIdentifier identifier) {
		Mandant mandant = MandantFactory.fromIdentifier(identifier);
		Gesuchsperiode gesuchsperiode = TestDataUtil.createGesuchsperiode1718(mandant);
		Gemeinde gemeinde = TestDataUtil.createGemeindeLondon(mandant);

		TestDataInstitutionStammdatenBuilder institution = new TestDataInstitutionStammdatenBuilder(gesuchsperiode);
		var testFall = new Testfall01_WaeltiDagmar(gesuchsperiode, true, gemeinde, institution);
		Gesuch gesuch = testFall.setupGesuch();
		verfuegen(gesuch);

		var bruennen = gesuch.getKindContainers().stream().findFirst().get().getBetreuungen().stream().findFirst().get();
		File pdf = generatePdf(bruennen, VerfuegungPdfGeneratorKonfiguration.builder()
			.FKJVTexte(true)
			.betreuungspensumAnzeigeTyp(BetreuungspensumAnzeigeTyp.NUR_PROZENT)
			.kontingentierungEnabledAndEntwurf(false)
			.stadtBernAsivConfigured(false)
			.isHoehereBeitraegeConfigured(false)
			.build(), "WaeltiDagmar.pdf");
		String text = PdfUnitTestUtil.getText(pdf);

		expect.scenario(identifier.name())
			.toMatchSnapshot(textForSnapshot(text));

		assertThat(text, stringContainsInOrder("Verfügung"));

	}

	@Test
	@SnapshotName("verfuegung-bruennen-hoeherer-beitrag")
	void testHoehererBeitrag() {
		Mandant mandant = MandantFactory.fromIdentifier(MandantIdentifier.SCHWYZ);
		Gesuchsperiode gesuchsperiode = TestDataUtil.createGesuchsperiode1718(mandant);
		Gemeinde gemeinde = TestDataUtil.createGemeindeLondon(mandant);

		TestDataInstitutionStammdatenBuilder institution = new TestDataInstitutionStammdatenBuilder(gesuchsperiode);
		var testFall = new Testfall01_WaeltiDagmar(gesuchsperiode, true, gemeinde, institution);
		Gesuch gesuch = testFall.setupGesuch();
		verfuegen(gesuch);

		var bruennen = gesuch.getKindContainers().stream().findFirst().get().getBetreuungen().stream().findFirst().get();
		File pdf = generatePdf(bruennen, VerfuegungPdfGeneratorKonfiguration.builder()
			.FKJVTexte(true)
			.betreuungspensumAnzeigeTyp(BetreuungspensumAnzeigeTyp.NUR_PROZENT)
			.kontingentierungEnabledAndEntwurf(false)
			.stadtBernAsivConfigured(false)
			.isHoehereBeitraegeConfigured(true)
			.build(), "WaeltiDagmar_hoehererBeitrag.pdf");
		String text = PdfUnitTestUtil.getText(pdf);

		expect.scenario(MandantIdentifier.SCHWYZ.name())
			.toMatchSnapshot(textForSnapshot(text));

		assertThat(text, stringContainsInOrder("Höherer", "Beitrag"));
	}

	private void verfuegen(Gesuch gesuch) {
		gesuch.getKindContainers().forEach(kindContainer -> kindContainer.getBetreuungen().forEach(betreuung -> {
			final Verfuegung verfuegung = new Verfuegung();
			final VerfuegungZeitabschnitt zeitabschnitt =
				new VerfuegungZeitabschnitt(gesuch.getGesuchsperiode().getGueltigkeit());
			zeitabschnitt.getRelevantBgCalculationResult().setBetreuungspensumProzent(BigDecimal.valueOf(80));
			zeitabschnitt.getRelevantBgCalculationResult().setHoehererBeitrag(BigDecimal.valueOf(10));
			verfuegung.setZeitabschnitte(List.of(zeitabschnitt));
			betreuung.setVerfuegung(verfuegung);
			betreuung.setBetreuungsstatus(Betreuungsstatus.VERFUEGT);
		}));
	}

	private File generatePdf(Betreuung betreuung, VerfuegungPdfGeneratorKonfiguration konfiguration, String fileName) {
		Gesuch gesuch = betreuung.extractGesuch();
		Gesuchsperiode gesuchsperiode = gesuch.getGesuchsperiode();

		// there are no easy-to-use Mandant rules for test cases, so we create our own...
		Verfuegung verfuegung = new Verfuegung();
		VerfuegungZeitabschnitt single =
			new VerfuegungZeitabschnitt(new DateRange(gesuchsperiode.getGueltigkeit().getGueltigAb()).withFullMonths());
		single.getRelevantBgCalculationResult().setEinkommensjahr(gesuchsperiode.getBasisJahr());

		verfuegung.setZeitabschnitte(Collections.singletonList(single));

		GemeindeStammdaten gemeindeStammdaten = TestDataUtil.createGemeindeStammdaten(gesuch.extractGemeinde());
		var generator =
			new VerfuegungPdfGeneratorVisitor(
				betreuung,
				gemeindeStammdaten,
				Art.NORMAL,
				konfiguration
			);
		Mandant mandant = gesuch.extractMandant();

		try {
			Path path = Path.of(getOutputPath(mandant.getMandantIdentifier()).toString(), fileName);
			generator.getVerfuegungPdfGeneratorForMandant(mandant).generate(Files.newOutputStream(path));

			return path.toFile();
		} catch (IOException | InvoiceGeneratorException e) {
			throw new IllegalStateException("Failed to generate PDF", e);
		}
	}

	@Nonnull
	private String textForSnapshot(String text) {
		return PATTERN.matcher(text.replaceAll(Constants.DATE_FORMATTER.format(LocalDate.now()), "<TODAY>"))
			.replaceAll("");
	}

	static Path getOutputPath(MandantIdentifier mandant) {
		return Path.of(PATH_PREFIX, mandant.name());
	}

}
