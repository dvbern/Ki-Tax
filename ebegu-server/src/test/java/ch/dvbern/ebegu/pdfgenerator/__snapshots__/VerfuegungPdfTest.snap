verfuegung-bruennen-hoeherer-beitrag[SCHWYZ]=[
Verfügung 
 
Für die Betreuung von Simon Wälti, geboren am 13.04.2014, gewähren wir Ihnen gestützt auf §§ 11 ff.
KiBeG und §§ 13 ff. KiBeV sowie aufgrund der finanziellen Verhältnisse (sh. Beilagen) Beiträge an die
Kinderbetreuung in folgender Höhe:
 
Referenznummer 17.000001.002.-1.1
Name Kind Simon Wälti
Angebot Kita / schulische Tagesstruktur
Institution Kita Aaregg
Gemeinde London
I II III IV V VI VII
Von Bis Betreuungspensum in % Effektive
Betreuungsko
sten [Fr.]
Berechnete
Beiträge [Fr.]
Abzüglich
Minimaltarif
Höherer
Beitrag
gemäss § 17
KiBeV
An Institution
geschuldeter
Beitrag [Fr.]Effektiv Anspruch Vergünstigt
01.08.2017 31.07.2018 80% 0.00 % 0.00% 0.00 0.00 0.00 10.00 0.00
Freundliche Grüsse
Sachbearbeitung
Rechtsmittelbelehrung
Gegen diesen Entscheid betreffend die Anspruchsberechtigung bzw. die Kostengutsprache kann innert
20 Tagen bei der Gemeinde London Einsprache erhoben werden. Die Einsprache muss einen Antrag und
eine Begründung enthalten. Die angefochtene Verfügung ist beizulegen. Einspracheentscheide können
innert 20 Tagen seit deren Zustellung an den Regierungsrat weitergezogen werden.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern

]


verfuegung-bruennen[APPENZELL_AUSSERRHODEN]=[
Verfügung 
 
Für die familienergänzende Betreuung von Simon Wälti, geboren am 13.04.2014, erhalten Sie gestützt auf
das Gesetz zur Förderung der familienergänzenden Kinderbetreuung (KibeG; bGS 415.31) und die
Kinderbetreuungsverordnung (KibeV; bGS 415.311) aufgrund Ihres Beschäftigungsgrades und Ihrer
finanziellen Verhältnisse (Beilage 1) Betreuungsbeiträge in nachfolgender Höhe:
 
Die Auszahlung der Betreuungsbeiträge erfolgt gemäss Ermächtigung direkt an die Betreuungsinstitution.
 
Bitte beachten Sie:
In der vorliegenden Verfügung werden das anspruchsberechtigte Betreuungspensum sowie der
Selbstbehalt festgelegt (Art. 12 KibeV). Sollte die Spalte "Beitragshöhe in Fr." bei Ihnen den Betrag
von Fr. 0.- aufweisen, bedeutet das nicht, dass Sie keine Beiträge erhalten. Diese Spalte wird in
diesem Fall erst nach Meldung der effektiven Betreuungsstunden und deren Kosten ergänzt. Sie
werden über die Höhe des zur Auszahlung gelangenden Beitrages informiert.
 
Die Erziehungsberechtigten haben der SOVAR alle Auskünfte zu erteilen, die für die Beurteilung des
Anspruchs notwendig sind. Wesentliche Änderungen der tatsächlichen oder rechtlichen Verhältnisse sind
der SOVAR unaufgefordert zu melden (Art. 9 KibeG). Insbesondere mitzuteilen sind gemäss Art. 13 Abs. 1
KibeV
-   Änderungen des Beschäftigungsgrads der Erziehungsberechtigten um 10 % oder mehr
-   Änderungen des massgebenden Einkommens um 20 % oder mehr
-   Wohnsitzwechsel
-   Änderungen der massgeblichen Familienverhältnisse.
 
Referenznummer 17.000001.002.-1.1
Name Kind Simon Wälti
Institution Kita Aaregg
Gemeinde London
Von Bis Pensum in Stunden Kosten in CHF Selbstbehalt in
%
Beitragshöhe in
%
Beitragshöhe in
CHFVereinbarte
BS
Anspruch
generell
Anspruch
konkret
01.08.2017 31.07.2018 80% 0.00 % 0.00% 0.00 0.00
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern
Freundliche Grüsse
Sachbearbeitung
Rechtsmittelbelehrung
Gegen diesen Entscheid kann innerhalb von 20 Tagen nach Zustellung beim Departement Gesundheit
und Soziales Appenzell Ausserrhoden, Kasernenstrasse 17, 9102 Herisau schriftlich Rekurs eingereicht
werden. Der Rekurs hat einen Antrag und eine Begründung zu enthalten, allfällige Beweismittel sind
genau zu bezeichnen. Der angefochtene Entscheid und soweit möglich auch die Beweismittel sind
beizulegen

]


verfuegung-bruennen[BERN]=[
Verfügung 
 
Für die Betreuung von Simon Wälti, geboren am 13.04.2014, gewähren wir Ihnen gestützt auf die
gesetzlichen Grundlagen1 sowie aufgrund Ihrer finanziellen Verhältnisse (Beilage 1) einen
Betreuungsgutschein mit nachfolgender Vergünstigung:
 
Referenznummer 17.000001.002.-1.1
Name Kind Simon Wälti
Angebot Tagesstätte für Kleinkinder
Institution Kita Aaregg
Gemeinde London
1 Verordnung vom 24. November 2021 über die Leistungsangebote der Familien-, Kinder- und Jugendförderung (FKJV; BSG 860.22)
I II III IV V VI VII VIII
Von Bis Betreuungspensum in % Kosten in
CHF
Berechneter
Gutschein in
CHF
Betreuungs-
gutschein in
CHF
Abzüglich
minimaler
Elternbeitrag
in CHF
An Institution
geschuldeter
Betrag in CHFEffektiv Anspruch Vergünstigt
01.08.2017 31.07.2018 80% 0.00 % 0.00% 0.00 0.00 0.00 0.00 0.00
Freundliche Grüsse
Sachbearbeitung
Rechtsmittelbelehrung
Gegen diese Verfügung kann innert 30 Tagen Beschwerde erhoben werden. Die Beschwerdefrist kann
nicht verlängert werden. Die Beschwerde ist an die Adresse Jugendamt, Nussbaumstrasse 21, 3006 Bern
zuzustellen. Sie muss (a) angeben, welche Entscheidung anstelle der angefochtenen Verfügung beantragt
wird; (b) aus welchen Gründen diese andere Entscheidung verlangt wird, (c) die Unterschrift der
beschwerdeführenden Partei oder der sie vertretenden Person enthalten. Der Beschwerdeschrift
beizulegen sind die Beweismittel, soweit sie greifbar sind, und die angefochtene Verfügung.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern

]


verfuegung-bruennen[LUZERN]=[
Bestätigung 
 
Für die Betreuung von Simon Wälti, geboren am 13.04.2014, gewähren wir Ihnen gestützt auf die
gesetzlichen Grundlagen1 sowie aufgrund Ihrer finanziellen Verhältnisse (Beilage 1) einen
Betreuungsgutschein mit nachfolgender Vergünstigung:
 
Referenznummer 17.000001.002.-1.1
Name Kind Simon Wälti
Angebot Tagesstätte für Kleinkinder
Institution Kita Aaregg
Gemeinde London
1 Verordnung vom 24. November 2021 über die Leistungsangebote der Familien-, Kinder- und Jugendförderung (FKJV; BSG 860.22)
Von Bis Betreuungspensum in % Kosten in CHF An Institution
geschuldeter Betrag
in CHFEffektiv Anspruch Vergünstigt
01.08.2017 31.07.2018 80% 0.00 % 0.00% 0.00 0.00
Freundliche Grüsse
Fachbearbeitung Betreuungsgutscheine
Rechtsmittelbelehrung
Wenn Sie mit der Berechnung der Gutscheinhöhe oder den Ausführungen zum Anspruch auf
Betreuungsgutscheine nicht einverstanden sind nehmen Sie mit uns Kontakt auf.
Darüber hinaus können Sie einen schriftlichen Entscheid mit Begründung (Verfügung) verlangen.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern

]


verfuegung-bruennen[SCHWYZ]=[
Verfügung 
 
Für die Betreuung von Simon Wälti, geboren am 13.04.2014, gewähren wir Ihnen gestützt auf §§ 11 ff.
KiBeG und §§ 13 ff. KiBeV sowie aufgrund der finanziellen Verhältnisse (sh. Beilagen) Beiträge an die
Kinderbetreuung in folgender Höhe:
 
Referenznummer 17.000001.002.-1.1
Name Kind Simon Wälti
Angebot Kita / schulische Tagesstruktur
Institution Kita Aaregg
Gemeinde London
I II III IV V VI VII
Von Bis Betreuungspensum in % Effektive
Betreuungskos
ten [Fr.]
Berechnete
Beiträge [Fr.]
Abzüglich
Minimaltarif
An Institution
geschuldeter
Beitrag [Fr.]Effektiv Anspruch Vergünstigt
01.08.2017 31.07.2018 80% 0.00 % 0.00% 0.00 0.00 0.00 0.00
Freundliche Grüsse
Sachbearbeitung
Rechtsmittelbelehrung
Gegen diesen Entscheid betreffend die Anspruchsberechtigung bzw. die Kostengutsprache kann innert
20 Tagen bei der Gemeinde London Einsprache erhoben werden. Die Einsprache muss einen Antrag und
eine Begründung enthalten. Die angefochtene Verfügung ist beizulegen. Einspracheentscheide können
innert 20 Tagen seit deren Zustellung an den Regierungsrat weitergezogen werden.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern

]


verfuegung-bruennen[SOLOTHURN]=[
Verfügung 
 
Für die Betreuung von Simon Wälti, geboren am 13.04.2014, gewähren wir Ihnen gestützt auf die
gesetzlichen Grundlagen1 sowie aufgrund Ihrer finanziellen Verhältnisse (Beilage 1) einen
Betreuungsgutschein mit nachfolgender Vergünstigung:
 
Referenznummer 17.000001.002.-1.1
Name Kind Simon Wälti
Angebot Tagesstätte für Kleinkinder
Institution Kita Aaregg
Gemeinde London
1 Verordnung vom 24. November 2021 über die Leistungsangebote der Familien-, Kinder- und Jugendförderung (FKJV; BSG 860.22)
I II III IV V VI VII VIII
Von Bis Betreuungspensum in % Kosten in
CHF
Berechneter
Gutschein in
CHF
Betreuungs-
gutschein in
CHF
Abzüglich
minimaler
Elternbeitrag
in CHF
An Institution
geschuldeter
Betrag in CHFEffektiv Anspruch Vergünstigt
01.08.2017 31.07.2018 80% 0.00 % 0.00% 0.00 0.00 0.00 0.00 0.00
Freundliche Grüsse
Sachbearbeitung
Rechtsmittelbelehrung
Gegen diese Verfügung kann innert 10 Tagen ab Zustellung bei der Beschwerdekommission der Stadt
Solothurn Beschwerde erhoben werden. Beschwerdeentscheide können an das Departement des Innern
weitergezogen werden. Die Beschwerde muss einen Antrag und eine Begründung enthalten. Der
Beschwerdeschrift beizulegen sind wenn möglich die Beweismittel und die angefochtene Verfügung.
Jugendamt
Nussbaumstrasse 21
3006 Bern
Telefon 031 123 12 12
info@bern.ch
www.bern.ch
Bern, <TODAY>
Dagmar Wälti
Testweg 10
3000 Bern

]